From 1b918594f2f7d192f4415764450f3ce44a553014 Mon Sep 17 00:00:00 2001
From: "Carsten Haitzler (Rasterman)" <raster@rasterman.com>
Date: Mon, 12 Jun 2017 11:14:16 +0900
Subject: [PATCH] elua - fix build for luajit2.1.0-beta3+

it seems luajit (and lua) broke api again... removed a #define from
their headers from 5.1->5.2 of lua (and seemingly luajit2.1 too). :(

this should fix T2728

@fix
---
 src/lib/elua/elua_private.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/lib/elua/elua_private.h b/src/lib/elua/elua_private.h
index 9f62c5f..35c4f85 100644
--- a/src/lib/elua/elua_private.h
+++ b/src/lib/elua/elua_private.h
@@ -28,6 +28,11 @@
 #include <lualib.h>
 #include <lauxlib.h>
 
+// FIX for luajit 2.1.0-beta3: see https://phab.enlightenment.org/T2728
+#ifndef luaL_reg
+# define luaL_reg luaL_Reg
+#endif
+
 #include "Elua.h"
 
 struct _Elua_State
-- 
2.7.4

